lib/CL/pocl_build.c

```diff
@@ -179,6 +172,15 @@ program_compile_dynamic_wg_binaries (cl_program program, int only_spmd_devices)
 
           cmd.command.run.kernel = kernel;
 
+	  /* First force generate a WG function with a dynamic global
+	     offset to ensure generality. */
+	  cmd.command.run.pc.global_offset[0] = 1;
+          device->ops->compile_kernel (&cmd, kernel, device);
+
+	  /* Then generate a specialized one with goffset 0. */
+	  cmd.command.run.pc.global_offset[0] =
+	    cmd.command.run.pc.global_offset[1] =
+	    cmd.command.run.pc.global_offset[2] = 0;
```

Comment:
The specialization interface is majorly refactored in a patch I'm working on, so I'm reluctant touching this too much in this PR. In that case the "dynamic binary" will be a generic one which will run with any kernel commands (including goffs > 0), basically for binary creation purposes, and then we specialize the WG functions for the launches as much as possible per command.

Reply:
After finishing my 2nd coffee cup I think I understood your comment better. Yes, the intention indeed is to include also the goffs0 specialized version to the binary since it seems to really common to not use the global offset.